modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -96,6 +93,9 @@ public PagingProviderProducer(PagingProvider<Object, T> delegate,
   @Override
   public List<T> produce() {
     List<T> page = performWithConnection(delegate::getPage);
+    if (isFirstPage) {
+      tryToMutateConfigurationStats(config, (MutableConfigurationStats::addOpenedStream));
```

Comment:
What if the first page is not produced right after the paging provider is created?